department-of-veterans-affairs/caseflow
# Block Special Case Movement for Legacy Appeal - Scenario1

Resolves [APPEALS-24904](https://jira.devops.va.gov/browse/APPEALS-24904)

### Description

 Given the User has selected the task action "case movement", cancel blocking tasks and advance to Judge" on a Legacy appeal with blocking tasks

### Acceptance Criteria:

- [ ] User will be presented with a Confirm cancellation and Reassign modal
- [ ] User will be required to select a VLJ from the menu option to reassign case to
- [ ] User will be required to provide instructions/justification for the case assignment to VLJ
- [ ] When VLJ selection is made and instructions are provided, the Cancel Task and Reassign button becomes enabled
- [ ] When User clicks on the Cancel Task and Reassign button, User will receive a success confirmation banner

### Testing Plan for Legacy Appeal

1. Go to Caseflow login and sign in as `BVARDUNKLE`
2. Navigate `https://voyager.caseflowdemo.com/queue/appeals/` and any legacy appeal in my case `http://127.0.0.1:3000/queue/appeals/LEGACYIH`

![la1](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/4ca22316-bcd5-403b-8faf-cc0f5fd59305)

3. Check the task tree at the beginning go to `http://127.0.0.1:3000/explain/appeals/LEGACYIH`
Must be like this.

![latree](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/4e2cc75c-33ae-4bfe-9ac9-a904782ca7fb)

4. Open the second dropdown and select `Case Movement: Cancel blocking task and advance to judge `
5. Complete the form and click on `Continue`

![la2](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/91110309-6979-4f57-845a-6dc8d8855505)

6. When clicking on `Continue` a new modal is open, in this case, need to select a new attorney, and provide some instructions, then the `Cancel Task and Reassign` bottom display enable

![la3](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/938046f1-de8f-407a-9fd6-03e8b9a2adcb)

7. When clicking on `Cancel Task and Reassign` if everything is fine, you see the Detail page with the new changes and a green banner shows up.

![la4](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/03794f0f-f79e-4b52-90d3-732f377f2f9b)

8. Check the task tree at this point again  `http://127.0.0.1:3000/explain/appeals/LEGACYIH`
Must be like this.

![legatree](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/f0fc64db-387e-4538-b5a3-1b762166b3c0)


### Testing Plan for Regular Appeal

1. Go to Caseflow login and sign in as `BVADWISE` intake user
2. Navigate `https://voyager.caseflowdemo.com/intake` and create an appeal of hearing type.
3. Add one issue to this appeal, and save it, save the number of this appeal too.
4. Then change the user to `BVARDUNKLE`
5. Navigate `https://voyager.caseflowdemo.com/search` and paste the number of appeal that you save before.
6. have to see a page like this

![ra1](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/92785fe3-3cb8-4f96-9182-05e605c90768)

7. Check the task tree at the beginning. Must be like this.

![ratree](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/ba5c89b0-770e-48b0-881b-a9dd8ea8830b)

8. Open the second dropdown and select `Case Movement: Cancel blocking task and advance to judge `
9. Complete the form and click on `Continue`

![ra2](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/b2e73e1d-c141-4cd9-a762-1cd50f4e2da8)

10. When clicking on `Continue` a new modal is open, in this case, need to select a new attorney, and provide some instructions, then the `Cancel Task and Reassign` bottom display enable

![ra3](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/91686566-f864-4f34-9553-d899a2ee5bb4)

11. When clicking on `Cancel Task and Reassign` if everything is fine, you see the Detail page with the new changes and a green banner shows up.

![ra4](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/9ed9a04f-e0b3-4826-9b3b-685c61053776)

12. Check the task tree at this point again .
Must be like this.

![ratree2](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/77483b29-5155-42f0-ad2f-54645436e82e)
